### ISSUE TITLE
Pressing the up arrow in console should put the cursor at the end of the previous command

### ISSUE THREAD

**Issue by joonro** [jupyterlab/jupyterlab#2346]
Hi,

Currently (`0.23.1`) when I press up arrow in a console, the previous command shows up, but the cursor remains on the beginning of the line. I think the cursor should be placed at the end of the line, just like the qt console. Thank you!

**Comment by ellisonbg** [comment]
What happens in the qtconsole if you press up again and there is multiple lines in the previous history. It should probably go up to the previous line (otherwise how do you edit the code?). But then you can't go back through history quickly by just hitting up. I think that is why we had it go to the beginning as then you can press up again and keep going in the history. Further clarification would help us figure out the right behavior. Thanks!

**Comment by ian-r-rose** [comment]
@mpacer and I played with this for a bit a couple of weeks ago and found three different behaviors, one for each of IPython terminal, qtconsole, and JupyterLab. We intended to start this conversation, but it seems we got sidetracked, so thanks for opening this @joonro. 
Behaviors:
1. IPython terminal: Pressing up goes to the end of the previous command.
2. QtConsole: Pressing up goes to the end of the first line of the previous command.
3. JupyterLab console: Pressing up goes to the beginning of the previous command.

The distinction between options one and two only appears on multi-line commands. I have a slight preference for option two, since it allows both for easy editing of previous one-liners and for quick scrolling through history.

**Comment by joonro** [comment]
I agree. I just tested this with QT console, and I found its behavior (pressing up goes to the end of the first line of the previous command (let's say this is the command `-1`) for both single- and multi-line commands) preferable. 

Also, If you press up one more time (to command `-2`) and then press down, then qtconsole puts the cursor at the _end_ of the last line of the command `-1` as well, making it easy to scroll back.
